src/components/file_upload.tsx

```diff
@@ -49,13 +55,18 @@ export class FileUpload extends React.Component<FileUploadProps, {}> {
                 type="file"
                 onChange={this._onChangeHandler}
                 disabled={disabled}
+                placeholder={placeholder}
             />
             <div className="layout layout--gutter-sm">
                 <div className="layout__item u-fill file-upload__input-container">
                     <label className={inputClasses} htmlFor={id}>{selectedFileName}</label>
                 </div>
                 <div className="layout__item u-fit">
-                    <label className='btn btn--primary' htmlFor={id}>{I18n.translate('file upload button', 'Choose file')}</label>
+                    <label className='btn btn--primary' htmlFor={id}>
+                        {I18n.translate('file upload button', 'Choose file')}
+                        {/* TODO: Check with Alan which classes we need to have the spinner keep the button size */}
```

Comment:
I saw some issues here where the text and spinner were both visible at the same time.

Reply:
@woosie10 indeed can you help on this one maybe? the button gets a different size when we hide the text.